### PR TITLE
Security

### PR DESCRIPTION
**Pull request by Sergey113222**

### CODE REVIEW

File: common/src/main/java/com/example/cinemabasegradle/dto/RoleDto.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import javax.validation.constraints.NotNull;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@SuppressFBWarnings(value = {"EI_EXPOSE_REP", "EI_EXPOSE_REP2"})
+public class RoleDto {
+    @NotNull(message = "Name cannot be null")
```
Comment (MaksimMartsiusheu) [inline common/src/main/java/com/example/cinemabasegradle/dto/RoleDto.java]: Maybe `@NotBlank` will be better

File: rest/src/main/java/com/example/cinemabasegradle/security/JwtTokenFilter.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                    SecurityContextHolder.getContext().setAuthentication(authentication);
+                }
+            }
+        } catch (AuthenticationException authenticationException) {
+            SecurityContextHolder.clearContext();
+            logger.debug("Failed to process authentication request", authenticationException);
+            this.authenticationEntryPoint.commence(request, response, authenticationException);
+            return;
+        }
+        filterChain.doFilter(request, response);
```
Comment (MaksimMartsiusheu) [inline rest/src/main/java/com/example/cinemabasegradle/security/JwtTokenFilter.java]: Mb let's add this line to `try` block?

File: security/src/main/java/com/example/cinemabasegradle/service/JwtUserDetailsService.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    private final UserRepository userRepository;
+    private final BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    public User loadUserByEmail(String email, String password) {
+        User user = userRepository.findByEmailAndActiveTrue(email)
+                .orElseThrow(() -> new IllegalArgumentException(MESSAGE));
+        log.info("In loadByEmail() - user with email: {} successfully loaded", email);
+
+        if (!bCryptPasswordEncoder.matches(password, user.getPassword())) {
+            throw new IllegalArgumentException(MESSAGE);
```
Comment (MaksimMartsiusheu) [inline security/src/main/java/com/example/cinemabasegradle/service/JwtUserDetailsService.java]: _"User doesn't exist"_  - I think that's not quite true. User exists, but passwords not match

**Review by MaksimMartsiusheu — CHANGES_REQUESTED** [review]

File: common/src/main/java/com/example/cinemabasegradle/dto/RoleDto.java (hunk 1 above)
Comment (Sergey113222) [reply common/src/main/java/com/example/cinemabasegradle/dto/RoleDto.java]: done

File: rest/src/main/java/com/example/cinemabasegradle/security/JwtTokenFilter.java (hunk 2 above)
Comment (Sergey113222) [reply rest/src/main/java/com/example/cinemabasegradle/security/JwtTokenFilter.java]: done

File: security/src/main/java/com/example/cinemabasegradle/service/JwtUserDetailsService.java (hunk 3 above)
Comment (Sergey113222) [reply security/src/main/java/com/example/cinemabasegradle/service/JwtUserDetailsService.java]: done

**Pull request merged**